pkg/client/client.go

```diff
@@ -1115,16 +1115,18 @@ func (c *Client) CreateOrUpdateClusterRoleBinding(crb *rbacv1.ClusterRoleBinding
 		return nil
 	}
 
-	required := crb.DeepCopy()
-	mergeMetadata(&required.ObjectMeta, existing.ObjectMeta)
```

Comment:
This would remove all assigned annotations and labels. Such an approach is not recommended as objects can be annotated by 3rd party and such annotations should persist throughout reconciliations.

Reply:
If we get here, some other entity already deleted and recreated the CRB (they didn't update it, because that's not possible, which is why we're here in the first place). We have no way of knowing whether they preserved our annotations; they may have even added their own. Doing a full reset seems like the safest option.

Reply:
What if 3rd party only updates CRB by adding annotation or label? What would happen to said metadata during cluster update? Asking because as far as I can see such data will be removed by  CMO during reconciliation.